server/service/src/sensor/berlinger.rs

```diff
@@ -0,0 +1,380 @@
+
+extern crate temperature_sensor;
+use temperature_sensor::*;
+
+pub fn get_breach_row_type(breach_type: &BreachType) -> TemperatureBreachRowType {
+    match breach_type {
+        BreachType::ColdConsecutive => TemperatureBreachRowType::ColdConsecutive,
+        BreachType::ColdCumulative => TemperatureBreachRowType::ColdCumulative,
+        BreachType::HotConsecutive => TemperatureBreachRowType::HotConsecutive,
+        BreachType::HotCumulative => TemperatureBreachRowType::HotCumulative,
+    }
+}
+
+pub fn get_matching_sensor_serial(
+    connection: &StorageConnection,
+    serial: &str,
+) -> Result<Vec<Sensor>, RepositoryError> {
+    SensorRepository::new(connection)
+        .query_by_filter(SensorFilter::new().serial(EqualFilter::equal_to(&serial)))
+}
+
+pub fn get_matching_sensor_log(
+    connection: &StorageConnection,
+    sensor_id: &str,
+    datetime: NaiveDateTime,
+) -> Result<Vec<TemperatureLog>, RepositoryError> {
+    let filter = TemperatureLogFilter::new()
+        .sensor(SensorFilter::new().id(EqualFilter::equal_to(sensor_id)))
+        .datetime(DatetimeFilter::equal_to(datetime));
+
+    TemperatureLogRepository::new(connection).query_by_filter(filter)
+}
+
+pub fn get_matching_sensor_breach_config(
+    connection: &StorageConnection,
+    description: &str,
+    breach_type: &TemperatureBreachRowType,
+) -> Result<Vec<TemperatureBreachConfig>, RepositoryError> {
+    let filter = TemperatureBreachConfigFilter::new()
+        .description(EqualFilter::equal_to(description))
+        .r#type(EqualFilter::equal_to_breach_type(&breach_type));
+
+    TemperatureBreachConfigRepository::new(connection).query_by_filter(filter)
+}
+
+pub fn get_matching_sensor_breach(
+    connection: &StorageConnection,
+    sensor_id: &str,
+    start_datetime: NaiveDateTime,
+    _end_datetime: NaiveDateTime,
+    breach_type: &TemperatureBreachRowType,
+) -> Result<Vec<TemperatureBreach>, RepositoryError> {
+    let filter = TemperatureBreachFilter::new()
+        .sensor(SensorFilter::new().id(EqualFilter::equal_to(sensor_id)))
+        .r#type(EqualFilter::equal_to_breach_type(&breach_type))
+        .start_datetime(DatetimeFilter::equal_to(start_datetime));
+
+    TemperatureBreachRepository::new(connection).query_by_filter(filter)
+}
+
+pub fn get_logs_for_sensor(
+    connection: &StorageConnection,
+    sensor_id: &str,
+) -> Result<Vec<TemperatureLog>, RepositoryError> {
+    TemperatureLogRepository::new(connection).query_by_filter(
+        TemperatureLogFilter::new()
+            .sensor(SensorFilter::new().id(EqualFilter::equal_to(sensor_id))),
+    )
+}
+
+pub fn get_breaches_for_sensor(
+    connection: &StorageConnection,
+    sensor_id: &str,
+) -> Result<Vec<TemperatureBreach>, RepositoryError> {
+    TemperatureBreachRepository::new(connection).query_by_filter(
+        TemperatureBreachFilter::new()
+            .sensor(SensorFilter::new().id(EqualFilter::equal_to(sensor_id)))
+            .acknowledged(false),
```

Comment:
Why just acknowledged ?

Also was wondering about what we do for acknowledgement of ongoing breaches, it's a question that we've asked, and I just asked again in a telegram chat (i've added you)

Reply:
I was thinking that by default you'd only be interested in breaches that hadn't been acknowledged (and similarly only breach configs that were active)